sql/2021/security/iframe_allow_directives.sql

```diff
@@ -0,0 +1,43 @@
+#standardSQL
+# usage of different directives for allow attribute on iframes
+CREATE TEMP FUNCTION getNumWithAllowAttribute(payload STRING) AS ((
+  SELECT
+    COUNT(0)
+  FROM
+    UNNEST(JSON_EXTRACT_ARRAY(JSON_EXTRACT_SCALAR(payload, '$._security'), "$.iframe-allow-sandbox")) AS iframeAttr
+  WHERE
+    JSON_EXTRACT_SCALAR(iframeAttr, '$.allow') IS NOT NULL
+));
+
+SELECT
+  client,
+  SPLIT(TRIM(allow_attr), ' ')[OFFSET(0)] AS directive,
+  total_iframes_with_allow,
+  COUNT(0) AS freq,
+  COUNT(0) / total_iframes_with_allow AS pct
+FROM (
+  SELECT
+    t._TABLE_SUFFIX as client,
+    JSON_EXTRACT_ARRAY(JSON_EXTRACT_SCALAR(payload, '$._security'), "$.iframe-allow-sandbox") AS iframeAttrs
+  FROM
+    `httparchive.pages.2021_08_01_*` AS t),
+  UNNEST(iframeAttrs) AS iframeAttr,
+  UNNEST(REGEXP_EXTRACT_ALL(JSON_EXTRACT_SCALAR(iframeAttr, '$.allow'), r'(?i)([^,;]+)')) AS allow_attr
+JOIN (
+  SELECT
+    t._TABLE_SUFFIX as client,
+    SUM(getNumWithAllowAttribute(payload)) AS total_iframes_with_allow
+  FROM
+    `httparchive.pages.2021_08_01_*` AS t
```

Comment:
You shouldn't need these `t`'s I don't think?

```suggestion
    _TABLE_SUFFIX as client,
    JSON_EXTRACT_ARRAY(JSON_EXTRACT_SCALAR(payload, '$._security'), "$.iframe-allow-sandbox") AS iframeAttrs
  FROM
    `httparchive.pages.2021_08_01_*`),
  UNNEST(iframeAttrs) AS iframeAttr,
  UNNEST(REGEXP_EXTRACT_ALL(JSON_EXTRACT_SCALAR(iframeAttr, '$.allow'), r'(?i)([^,;]+)')) AS allow_attr
JOIN (
  SELECT
    _TABLE_SUFFIX as client,
    SUM(getNumWithAllowAttribute(payload)) AS total_iframes_with_allow
  FROM
    `httparchive.pages.2021_08_01_*`
```

Reply:
Ah it was the `as` being in lowercase it was complaining about - updated suggestion above.